### PR TITLE
Open api emitter

### PR DESCRIPTION
**Pull request by wilmveel**

### CODE REVIEW

**Review by jerrevanveluw — CHANGES_REQUESTED** [review]

File: src/converter/openapi/build.gradle.kts [hunk 1]
```diff
@@ -41,6 +42,7 @@ kotlin {
         }
         commonTest {
             dependencies {
+                implementation("com.goncalossilva:resources:0.4.0")
```
[the remark this answers is not in the document]
Comment (wilmveel) [reply src/converter/openapi/build.gradle.kts:45]: This plugin does it also fixes alle the paths for different targets

File: src/converter/openapi/src/commonMain/kotlin/community/flock/wirespec/openapi/v2/OpenApiV2Emitter.kt [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+import community.flock.wirespec.compiler.core.parse.Enum
+import community.flock.wirespec.compiler.core.parse.Refined
+import community.flock.wirespec.compiler.core.parse.Type
+import kotlinx.serialization.json.JsonPrimitive
+import community.flock.kotlinx.openapi.bindings.v2.Type as OpenApiType
+
+class OpenApiV2Emitter {
+
+    fun emit(ast: AST): SwaggerObject {
+        return SwaggerObject(
```
[the remark this answers is not in the document]
Comment (wilmveel) [reply src/converter/openapi/src/commonMain/kotlin/community/flock/wirespec/openapi/v2/OpenApiV2Emitter.kt:27]: done

File: src/converter/openapi/src/commonMain/kotlin/community/flock/wirespec/openapi/v2/OpenApiV2Emitter.kt [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+                                    true -> SchemaObject(
+                                        type = OpenApiType.ARRAY,
+                                        items = content.reference.emit()
+                                    )
+                                }
+                            }
+                    )
+                }
+        )
+        return operationObject
```
[the remark this answers is not in the document]
Comment (wilmveel) [reply src/converter/openapi/src/commonMain/kotlin/community/flock/wirespec/openapi/v2/OpenApiV2Emitter.kt:120]: done

File: src/converter/openapi/src/commonMain/kotlin/community/flock/wirespec/openapi/v2/OpenApiV2Emitter.kt [hunk 4]
```diff
@@ -0,0 +1,186 @@
...
+            is Type.Shape.Field.Reference.Custom -> ReferenceObject(ref = Ref("#/definitions/${value}"))
+            is Type.Shape.Field.Reference.Primitive -> SchemaObject(
+                type = type.emitType(),
+            )
+
+            is Type.Shape.Field.Reference.Any -> error("Cannot map Any")
+            is Type.Shape.Field.Reference.Unit -> error("Cannot map Unit")
+        }
+
+        if (isIterable) {
```
[the remark this answers is not in the document]
Comment (wilmveel) [reply src/converter/openapi/src/commonMain/kotlin/community/flock/wirespec/openapi/v2/OpenApiV2Emitter.kt:159]: done

File: src/converter/openapi/src/commonMain/kotlin/community/flock/wirespec/openapi/v2/OpenApiV2Emitter.kt [hunk 5]
```diff
@@ -0,0 +1,186 @@
...
+
+    private fun Type.Shape.Field.Reference.Primitive.Type.emitType(): OpenApiType = when (this) {
+        Type.Shape.Field.Reference.Primitive.Type.String -> OpenApiType.STRING
+        Type.Shape.Field.Reference.Primitive.Type.Integer -> OpenApiType.INTEGER
+        Type.Shape.Field.Reference.Primitive.Type.Number -> OpenApiType.NUMBER
+        Type.Shape.Field.Reference.Primitive.Type.Boolean -> OpenApiType.BOOLEAN
+    }
+
+    private fun Type.Shape.Field.Reference.emitType(): OpenApiType =
+        when (isIterable) {
```
[the remark this answers is not in the document]
Comment (wilmveel) [reply src/converter/openapi/src/commonMain/kotlin/community/flock/wirespec/openapi/v2/OpenApiV2Emitter.kt:177]: done

File: src/converter/openapi/build.gradle.kts (hunk 1 above)
[the remark this answers is not in the document]
Comment (jerrevanveluw) [reply src/converter/openapi/build.gradle.kts:45]: Maybe fix it later then. Only used for tests I saw.

**Review by jerrevanveluw — APPROVED** [review]

**Pull request merged**